EE.C

```diff
@@ -0,0 +1,478 @@
+                    lep2.SetPtEtaPhiE(Lep_pt[1], Lep_eta[1], Lep_phi[1], Lep_E[1]);
+                }
+
+                else if(Nbjet == 1){
+                    bjet1.SetPtEtaPhiE(bJet_pt[0], bJet_eta[0], bJet_phi[0], bJet_E[0]);
+                    lep1.SetPtEtaPhiE(Lep_pt[0], Lep_eta[0], Lep_phi[0], Lep_E[0]);
+
+                    bjet2.SetPtEtaPhiE(Jet_pt[Jet2_nonB], Jet_eta[Jet2_nonB], Jet_phi[Jet2_nonB], Jet_E[Jet2_nonB]);
+                    lep2.SetPtEtaPhiE(Lep_pt[1], Lep_eta[1], Lep_phi[1], Lep_E[1]);
+                }
+
+                float lep1_jet1_mass = ((lep1)+(bjet1)).M();
+                float lep1_jet2_mass = ((lep1)+(bjet2)).M();
+                float lep2_jet1_mass = ((lep2)+(bjet1)).M();
+                float lep2_jet2_mass = ((lep2)+(bjet2)).M();
+
+                float SBM1 = 0.;
+                float SBM2 = 0.;
+                if(abs(lep1_jet1_mass - lep2_jet2_mass) < abs(lep2_jet1_mass - lep1_jet2_mass)){
+                    SBM1 = lep1_jet1_mass;
+                    SBM2 = lep2_jet2_mass;
+                }
+                else{
+                    SBM1 = lep2_jet1_mass;
+                    SBM2 = lep1_jet2_mass;
+                }
+
+                //Fill max SBM
+                //if(SBM1>170. || SBM2>170.){//>170 cut
+                    if(SBM1<SBM2)SBM_DY->Fill(SBM2);
+                    else SBM_DY->Fill(SBM1);
+
+
+
+                    //METvsMmm
+                    //if(Nnon_bjet == 1 && Nbjet == 0){
+                    float Mmm = ((lep1)+(lep2)).M();
+                    float MET_t = MET[0];
+                    float METvsMmm = MET_t/Mmm ;
+                    //if(METvsMmm<0.2){ // <0.2 cut
+                        METvsMmm_DY->Fill(METvsMmm);
+
+
+                        MET_DY->Fill(MET_t);
+                        dilep_mass_DY->Fill(Mmm);
+                        //}
+                        //HT-LT
+                        float DHTLT = -999.;
+                        if(Nbjet == 2){
+                            DHTLT = (bJet_pt[0]+bJet_pt[1]) - (Lep_pt[0]+Lep_pt[1]);
+                        }
+                        else if(Nbjet == 1){
+                            DHTLT = (bJet_pt[0]+Jet2_nonB_PT) - (Lep_pt[0]+Lep_pt[1]);
+                        }
+                        else continue;
+
+                        //if(DHTLT<0){//<0 cut
+                            HTLT_DY->Fill(DHTLT);
+                        //}
+                //}
+                //}
+                float Mm_1 = (lep1).M();
+                float Mm_2 = (lep2).M();
+                Mass_DY->Fill(Mm_1);
+                Mass_DY->Fill(Mm_2);
+            }
+            //leading jet & 2nd leading jet pT    
+            if(secjet_pT == -999) secjet_pT = leadjet_pT;
+            LeadJetpT_DY->Fill(leadjet_pT);
+            SecJetpT_DY->Fill(secjet_pT);
+
+	  }
+      }
+      }
+      }
+      
+      
+   }
+  
+   run_DY->Write();
```

Comment:
For such purposes, it's simplest to make a common vector of histograms in which you push back all the histograms on declaration once. In C++10x standard, you can even do this by an explicit list from the get go.
E.g. with histograms `TH1F *A, *B, *C;` you can do `vector<TH1F*> hists={A,B,C};` and at the end of the file just do:
`for(unsigned i=0; i<hists.size(); ++i) hists[i]->Write();`